client/state/connectionMiddleware/middleware.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import uniqueId from 'lodash/utility/uniqueId';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isOffline,
+	isOnline
+} from 'state/application/selectors';
+import {
+	OFFLINE_QUEUE_ACTION,
+	CONNECTION_RESTORED,
+	OFFLINE_QUEUE_REMOVE,
+} from 'state/action-types';
+const ACTION_TIMEOUT = 10000;
+
+const connections = {};
+
+let actionTimeout = null;
+
+//Prob should be in actions
+function queueAction( action ) {
+	return {
+		type: OFFLINE_QUEUE_ACTION,
+		id: uniqueId(),
+		dispatchedAt: ( new Date() ).getTime(),
+		action: action
```

Comment:
I think we are going to need a system to give priority to certain actions (maybe following WP core numbered priority system) that should go first. The thing is if in some cases some of these actions have dependencies on lower priority actions to occur first.

```
priority: 1 || 10
```